afs/afs-core/src/main/java/com/powsybl/afs/Node.java

```diff
@@ -26,8 +26,14 @@ protected Node(FileCreationContext context, int codeVersion, boolean folder) {
     }
 
     public void delete() {
-        storage.deleteNode(info.getId());
-        storage.flush();
+        if (isFolder()) {
+            Folder folderToDelete = (Folder) this;
+            if (folderToDelete.getChildren().isEmpty()) {
+                storage.deleteNode(info.getId());
+                storage.flush();
```

Comment:
flush should be done just one time when last folder is deleted

Reply:
```  if (isFolder()) {
            Folder folderToDelete = (Folder) this;
            if (folderToDelete.getChildren().isEmpty()) {
                storage.deleteNode(info.getId());
            }
        }
           storage.flush();
```

Reply:
which hierarchy? it only concerns non empty folders